.github/workflows/release-validate.yml

```diff
@@ -0,0 +1,51 @@
+name: release-validation
```

Comment:
I'm not sure why we could need this workflow? Do we not trust GH events?

Reply:
This validates if merging a PR will not create an invalid version. For example, a PR with a new feature targeting a branch that should only receive patches/fixes. Without this check, if the PR gets merged then the invalid commit will block the next release from happening.